src/components/tabBar/TabBarItem.tsx

```diff
@@ -1,147 +1,166 @@
+const HORIZONTAL_PADDING = Constants.isTablet ? Spacings.s7 : Spacings.s5;
+
+interface Props extends ThemeComponent {
   /**
    * icon of the tab
    */
-  icon?: number,
+  icon?: number;
   /**
    * icon tint color
    */
-  iconColor?: string,
+  iconColor?: string;
   /**
    * icon selected tint color
    */
-  iconSelectedColor?: string,
+  iconSelectedColor?: string;
   /**
    * label of the tab
    */
-  label?: string,
+  label?: string;
   /**
    * custom label style
    */
-  labelStyle?: TextStyle,
+  labelStyle?: StyleProp<TextStyle>;
   /**
    * Badge component props to display next the item label
    */
-  badge?: BadgeProps,
+  badge?: BadgeProps; //TODO: remove after deprecation
+  badgeProps?: BadgeProps;
   /**
-   * maximum number of lines the label can break
+   * maximun number of lines the label can break
    */
-  maxLines?: number,
+  maxLines?: number;
   /**
    * custom selected tab label style
    */
-  selectedLabelStyle: TextStyle,
+  selectedLabelStyle?: StyleProp<TextStyle>;
   /**
    * whether the tab is selected or not
    */
-  selected?: boolean,
+  selected?: boolean;
   /**
    * whether the tab will have a divider on its right
    */
-  showDivider?: boolean,
+  showDivider?: boolean;
   /**
    * A fixed width for the item
    */
-  width?: number,
+  width?: number;
+  /**
+   * tabBar's background color
+   */
+  backgroundColor?: string;
   /**
    * ignore of the tab
    */
-  ignore?: boolean,
+  ignore?: boolean;
   /**
    * callback for when pressing a tab
    */
-  onPress?: (props: any) => void,
+  onPress?: () => void;
   /**
    * whether to change the text to uppercase
    */
-  uppercase?: boolean,
+  uppercase?: boolean;
   /**
    * Apply background color on press for TouchableOpacity
    */
-  activeBackgroundColor?: string,
-  indicatorStyle?: ViewStyle, // for inner use
-  style?: ViewStyle,
-  testID?: string
-};
-
-export type State = {
-  indicatorOpacity?: Animated.Value
+  activeBackgroundColor?: string;
+  accessibilityLabel?: string;
+  indicatorStyle?: StyleProp<ViewStyle>; // for inner use
+  style?: ViewStyle;
+  testID?: string;
 }
 
+interface State {
+  indicatorOpacity: Animated.Value;
+  selected: boolean;
+}
 
-const INDICATOR_BG_COLOR = Colors.blue30;
-const INDICATOR_HEIGHT = 2;
-const INDICATOR_SPACINGS = 16;
+export type TabBarItemProps = Props;
 
 /**
  * @description: TabBar.Item, inner component of TabBar for configuring the tabs
- * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabBarScreen.js
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabBarScreen.tsx
  * @extends: TouchableOpacity
  * @extendsLink: https://facebook.github.io/react-native/docs/touchableopacity
  */
-class TabBarItem extends PureComponent<TabBarItemProps, State> {
+export default class TabBarItem extends PureBaseComponent<Props, State> {
```

Comment:
Shouldn't we take this chance to migrate to asBaseComponent?

Reply:
Yes. I'll migrate it, no reason not to

Reply:
Done